### PR TITLE
Rework front end knockout

### PR DESCRIPTION
**Pull request by Eirika**

### CODE REVIEW

File: website/static/css/style.less [hunk 1]
```diff
@@ -465,26 +620,22 @@ button > .glyphicon, a > .glyphicon {
...
 	top: 0;
-	height: 390px;
 	display: none;
 	z-index: 1;
 	&.player-child-no-music {
-		display: block;
+		display: table;
 		z-index: 0;
-		background: url('../img/cover_center_small.jpg') no-repeat;
+		//background: url('../img/cover_center_small.jpg') no-repeat;
```
Comment (Amoki) [inline website/static/css/style.less]: dead

File: website/static/js/knockout/customBinding.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      }
+      $(element).addClass('selectpicker').selectpicker();
+    }
+  },
+  update: function(element, valueAccessor, allBindingsAccessor) {
+    if($(element).is('select')) {
+      var selectPickerOptions = allBindingsAccessor().selectPickerOptions;
+      if(typeof selectPickerOptions !== 'undefined' && selectPickerOptions !== null) {
+        var options = selectPickerOptions.optionsArray;
+        isDisabled = selectPickerOptions.disabledCondition || false,
```
Comment (Amoki) [inline website/static/js/knockout/customBinding.js]: var

File: website/static/js/knockout/customBinding.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        resetOnDisabled = selectPickerOptions.resetOnDisabled || false;
```
Comment (Amoki) [inline website/static/js/knockout/customBinding.js]: var

File: website/static/js/knockout/customBinding.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+      $(element).selectpicker('refresh');
+    }
+  }
+};
+
+ko.subscribable.fn.trimmed = function() {
+  return ko.computed({
+    read: function() {
+      if(this()) {
```
Comment (Amoki) [inline website/static/js/knockout/customBinding.js]: ???

File: website/static/js/knockout/models.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  this.music_id = ko.observable(data.music_id);
+  this.name = ko.observable(data.name);
+  this.url = ko.observable(data.url);
+  this.room = ko.observable(data.room);
+  this.date = ko.observable(data.date);
+  this.duration = ko.observable(data.duration);
+  this.thumbnail = ko.observable(data.thumbnail);
+  this.count = ko.observable(data.count || data.views);
+  this.last_play = ko.observable(data.last_play);
+  this.dead_link = ko.observable(data.dead_link);
```
Comment (Amoki) [inline website/static/js/knockout/models.js]: no more

File: website/static/js/knockout/models.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  this.current_time_left = ko.observable(data.current_time_left);
+  this.current_time_past = ko.observable(data.current_time_past);
+  this.current_time_past_percent = ko.observable(data.current_time_past_percent);
+  this.can_adjust_volume = ko.observable(data.can_adjust_volume);
+  this.shuffle = ko.observable(data.shuffle);
+  this.count_left = ko.observable(data.count_left);
+
+  this.currentMusic = ko.observable();
+  if(data.current_music) {
+    this.currentMusic = new Music(data.current_music);
```
Comment (Amoki) [inline website/static/js/knockout/models.js]: condition ternaire ?

File: website/static/js/lib/common_functions.js [hunk 7]
```diff
@@ -47,26 +75,44 @@ function modalConfirm(target) {
...
+jQuery.fn.opacityToggle = function() {
+  return this.css('opacity', function(i, opacity) {
+    return (opacity === 1) ? 0 : 1;
   });
-});
+};
+
+function humanizeSeconds(s) {
+  var fm = [
+  Math.floor(s / 60) % 60,
```
Comment (Amoki) [inline website/static/js/lib/common_functions.js]: indent

File: website/static/js/lib/jquery.countTo.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+      from:            this.$element.data('from'),
+      to:              this.$element.data('to'),
+      speed:           this.$element.data('speed'),
+      refreshInterval: this.$element.data('refresh-interval'),
+      decimals:        this.$element.data('decimals')
+    };
+
+    var keys = Object.keys(options);
+
+    for (var i in keys) {
```
Comment (Amoki) [inline website/static/js/lib/jquery.countTo.js]: for (let i in keys) {

File: website/static/js/lib/jquery.countTo.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    return options;
+  };
+
+  CountTo.prototype.update = function () {
+    this.value += this.increment;
+    this.loopCount++;
+
+    this.render();
+
+    if (typeof(this.options.onUpdate) == 'function') {
```
Comment (Amoki) [inline website/static/js/lib/jquery.countTo.js]: it can be something else ?

File: website/static/js/lib/common_functions.js (hunk 7 above)
Comment (Eirika) [inline website/static/js/lib/common_functions.js]: ??

Comment (Amoki) [inline website/static/js/lib/common_functions.js]: each elem of the list should be intended compared to "["

File: website/static/js/remote/viewModels.js [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+        self.room(new Room(allData));
+        $('.btn-skip-music').prop('disabled', false);
+        modalConfirm($('#modal-next-music'));
+      },
+      error: logErrors,
+    });
+  };
+
+  self.postPlaylistSort = function(pk, action, target) {
+    target = (typeof target === 'undefined') ? '' : target;
```
Comment (Amoki) [inline website/static/js/remote/viewModels.js]: target = target || ''

File: website/static/js/remote/viewModels.js [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+    $('.overlay-playlist').show();
+    $.ajax({
+      url: '/playlist/' + pk + '/' + action + '/' + target,
```
Comment (Amoki) [inline website/static/js/remote/viewModels.js]: currenly /endpoint/ and /endpoint are differents :/

**Pull request merged**